### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by NouemanKHAL**
Improve the README documentation, and include the new CAPI metadata collection feature.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,6 +25,38 @@ Once it is available in your Cloud Foundry environment, configure your applicati
...
+    DD_SERVICE: <SERVICE_NAME>
+    DD_VERSION: <VERSION>
+  metadata:
+    labels:
+      tags.datadoghq.com/env: <ENV_NAME>
+      tags.datadoghq.com/service: <SERVICE_NAME>
+      tags.datadoghq.com/version: <VERSION>
+```
+
+The `tags.datadoghq.com` prefix is part of the Agent Autodiscovery notation as described in [Basic Agent Autodiscovery Documentation](https://docs.datadoghq.com/getting_started/containers/autodiscovery).
```
Comment (sarah-witt) [inline README.md:47]: Can we also mention how this autodiscovery prefix is a way to set "normal" tags as well?

File: README.md [hunk 2]
```diff
@@ -25,6 +25,38 @@ Once it is available in your Cloud Foundry environment, configure your applicati
 
 ### Configuration
 
+#### Unified Service Tagging
+
+> This feature requires the Datadog Cluster Agent to be installed. 
+See [Datadog Cluster Agent BOSH Release](https://github.com/DataDog/datadog-cluster-agent-boshrelease).
+
+Unified service tagging ties Datadog telemetry together through using three reserved tags: `env`, `service`, and `version`. In Cloud Foundry, they are set through the application labels/annotations.
```
Comment (sarah-witt) [inline README.md:33]: Can you mention how they are also set through environment variables?

File: README.md [hunk 3]
```diff
@@ -56,8 +88,12 @@ env:
 
 The following environment variables are used to configure log collection.
```
Comment (sarah-witt) [inline README.md:89]: Thoughts on if we make this a section about all available environment variables, instead of just log collection?

**Review by sarah-witt — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -25,6 +25,38 @@ Once it is available in your Cloud Foundry environment, configure your applicati
...
+You can find more information in the [Unified Service Tagging Documentation](https://docs.datadoghq.com/getting_started/tagging/unified_service_tagging).
+
+#### Application Metadata collection
+
+> This feature requires both the Datadog Agent and the Datadog Cluster Agent to be installed. 
+See [Datadog Agent BOSH Release](https://github.com/DataDog/datadog-cluster-agent-boshrelease) and [Datadog Cluster Agent BOSH Release](https://github.com/DataDog/datadog-cluster-agent-boshrelease).
+
+You can enable the collection of your application metadata (labels and annotations) as tags in your application logs, traces and metrics by setting the environment variable `DD_ENABLE_CAPI_METADATA_COLLECTION` to `true`.
+
+__Note__: Enabling this feature might trigger a restart of the Datadog Agent when the application metadata are updated, depending on the `cloud_foundry_api.poll_interval` on the Datadog Cluster Agent. On average, it takes around 20 seconds to restart the agent.
```
[the remark this answers is not in the document]
Comment (NouemanKHAL) [reply README.md:58]: Metadata is actually a plural noun, metadatum being its singular form.

File: README.md (hunk 3 above)
Comment (NouemanKHAL) [reply README.md:89]: This will be handled by the documentation team.

**Review by sarah-witt — APPROVED** [review]

**Pull request merged**